packages/app/app/mint/page.tsx

```diff
@@ -1,61 +1,170 @@
 "use client";
 
-import { useState } from "react";
+import { useEffect, useMemo, useState } from "react";
+import { useAccount, useNetwork } from "wagmi";
 import Image from "next/image";
 import { StacklyBetaNFTImg } from "@/public/assets";
+import {
+  ChainId,
+  getNftWhitelistAddress,
+  getWhitelist,
+  nftWhitelistBalanceOf,
+  nftWhitelistMaxSupply,
+  nftWhitelistMint,
+  nftWhitelistTotalSupply,
+} from "@stackly/sdk";
 import { BodyText, Button, ButtonLink, HeadingText } from "@/ui";
+import { useEthersProvider, useEthersSigner } from "@/utils/ethers";
+import { ConnectButton, DialogConfirmTransactionLoading } from "@/components";
+import { ModalId, useModalContext } from "@/contexts";
 
 export default function Page() {
-  const [hasMinted, setHasMinted] = useState(false);
+  const { chain } = useNetwork();
+  const signer = useEthersSigner();
+  const { address, isDisconnected, isConnected } = useAccount();
+  const publicClient = useEthersProvider();
+  const { closeModal, isModalOpen, openModal } = useModalContext();
+
+  const [mintedAmount, setMintedAmount] = useState<string>("-");
+  const [maxSupply, setMaxSupply] = useState<string>("-");
+  const [isNFTHolder, setIsNFTHolder] = useState<boolean>(false);
+  const [error, setError] = useState<string | undefined>();
+
+  const nftWhitelist = useMemo(
+    () => getWhitelist(getNftWhitelistAddress(ChainId.GNOSIS), publicClient),
+    [publicClient]
+  );
+
+  useEffect(() => {
+    const fetchData = async () => {
+      setError(undefined);
+
+      Promise.allSettled([
+        nftWhitelistTotalSupply(nftWhitelist),
+        nftWhitelistMaxSupply(nftWhitelist),
+      ])
+        .then(([newMintedAmount, newMaxSupply]) => {
+          if (newMintedAmount.status === "fulfilled")
+            setMintedAmount(newMintedAmount.value.toString());
+          if (newMaxSupply.status === "fulfilled")
+            setMaxSupply(newMaxSupply.value.toString());
+        })
+        .catch((e) =>
+          console.error("Error while fetching NFT contract info: ", e)
```

Comment:
should we `setError` here to give feedback to user?

Reply:
No, this error isn't related to minting the NFT. User can still mint the NFT if this throws an error.